Kernel/Interrupts/InterruptManagement.cpp

```diff
@@ -98,6 +98,19 @@ u8 InterruptManagement::get_irq_vector(u8 mapped_interrupt_vector)
     return mapped_interrupt_vector;
 }
 
+RefPtr<IRQController> InterruptManagement::get_responsible_irq_controller(IRQControllerType controller_type, u8 interrupt_vector)
+{
+    if (m_interrupt_controllers.size() == 1 && m_interrupt_controllers[0]->type() == IRQControllerType::i8259) {
+        return m_interrupt_controllers[0];
+    }
+    for (auto& irq_controller : m_interrupt_controllers) {
+        if (irq_controller->gsi_base() <= interrupt_vector)
+            if (irq_controller->type() == controller_type)
```

Comment:
Couldn't we just combine these two `if` statements with a one `if` statement with &&:
```cpp
        if (irq_controller->gsi_base() <= interrupt_vector && irq_controller->type() == controller_type)
```